### ISSUE TITLE
Fix false positives for math operations in function-whitespace-after

### ISSUE THREAD

**Issue by laurirasanen** [stylelint/stylelint#5071]
> Clearly describe the bug

Auto-fix for `function-whitespace-after` can cause invalid CSS properties when doing math with parantheses and using node-sass to compile SCSS to CSS.

> Which rule, if any, is the bug related to?

`function-whitespace-after`

> What code is needed to reproduce the bug?

Initial:  
```scss
$sample: 100;

a {
  z-index: (sample*2)-1;
}
```

Auto-fix will format this to:  
```scss
$sample: 100;

a {
  z-index: (sample*2) -1;
}
```
node-sass will output this as `z-index: 200 -1;` which is an invalid prop at least in Firefox 83.0 and Chrome 87.0.4280.88.  
The `-` sign should have a space on both sides or neither for node-sass to output `z-index: 199;` as expected.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-whitespace-after": "always"
  }
}
```

> Which version of stylelint are you using?

`13.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

[stylelint/vscode-stylelint](https://github.com/stylelint/vscode-stylelint) VSCode extension.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

SCSS with node-sass.

> What did you expect to happen?

Formatting should add a space after the `-` sign as well or not add any.

> What actually happened (e.g. what warnings or errors did you get)?

Formatting only places a space only before the `-` sign, causing an invalid property.

**Comment by jeddy3** [comment]
@laurirasanen Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilAvAtAJaawAe8KAFACQBuSATgJT4CMA3FgL4gA0IAZwCemAC5JSCcAIF9wETADNCAcykZsKdCAYBXADYwB2ihpw5ti3ZjCjCC-AHcAFoVFGADkjAx8SRe4MJlrI+o5IQsYgWDhc3CBcQA) with:

```scss
a {
  z-index: ($var)-1;
}
```

I believe the bug can be fixed by using the [`isStandardSyntaxFunction` util](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxFunction.js). However, `function-whitespace-after` is one of the older rules and uses [style-search](https://www.npmjs.com/package/style-search) rather than [postcss-value-parser](https://github.com/TrySound/postcss-value-parser) to identify functions within values (which the util relies on). The rule may need to be rewritten to use postcss-value-parser first, or there may be another way of ignoring these math constructs without a rewrite.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by yuzhiyao** [comment]
I have a similar problem；

version :  "stylelint": "^13.12.0",

Initial:
background-color: color(~`colorpalette("@{primary-color}", 5)`);

Auto-fix will format this to: 
background-color: color(~`colorpalette("@{primary-color}", 5) `);

There is an extra space after formatting；（function-whitespace-after）

The compiler returned an error [LessError: JavaScript evaluation error: 'ReferenceError: colorpalette is not defined']

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
